superset/viz.py

```diff
@@ -197,13 +197,11 @@ def get_df(self, query_obj=None):
         # be considered as the default ISO date format
         # If the datetime format is unix, the parse will use the corresponding
         # parsing logic.
-        if df is None or df.empty:
-            return pd.DataFrame()
-        else:
+        if not df.empty:
```

Comment:
You removed the check for `df is None` does that mean df won't be None? Will we still be passing an empty data frame if there's no data?

Reply:
The check was only there because we were mocking the pandas.DataFrame. The data is obtained from [pandas.read_sql_query](https://pandas.pydata.org/pandas-docs/version/0.22.0/generated/pandas.read_sql_query.html) which returns a pandas.DataFrame object.